### PR TITLE
[CELEBORN-556][BUG] ReserveSlot should not use default RPC time out since register shuffle max timeout is network timeout

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Currently, reserve slot uses same timeout with default RPC_ASK_TIMEOUT, 
we know that the register shuffle's max time out should be the min value of (RPC_ASK_TIMEOUT, REGISTER_SHUFFLE_RPC_ASK_TIMEOUT), if one worker is being reserving slots stuck, the the register shuffle process time must greater than the min value of (RPC_ASK_TIMEOUT, REGISTER_SHUFFLE_RPC_ASK_TIMEOUT).

The ShuffleClient side will throw register shuffle timeout exception,  so we should support customize the reserve slots rpc timeout, if worker stuck, we can give this process a chance to retry according to the timeout configuration

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @RexXiong

**Comment by AngersZhuuuu** [comment]
ping @RexXiong @waitinfuture , Could you take a look, this issue cause register shuffle failed.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -654,6 +654,29 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
   def quotaManagerClass: String = get(QUOTA_MANAGER)
   def quotaConfigurationPath: Option[String] = get(QUOTA_CONFIGURATION_PATH)
 
+  // //////////////////////////////////////////////////////
+  //                Shuffle Client RPC                   //
+  // //////////////////////////////////////////////////////
+  def reserveSlotsRpcTimeout: RpcTimeout =
+    new RpcTimeout(get(RESERVE_SLOTS_RPC_TIMEOUT).milli, RESERVE_SLOTS_RPC_TIMEOUT.key)
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:661]: whitespace

**Review by waitinfuture — APPROVED** [review]
LGTM except a nit

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -2493,6 +2501,14 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("5s")
 
+  val RESERVE_SLOTS_RPC_TIMEOUT: ConfigEntry[Long] =
+    buildConf("celeborn.rpc.reserveSlots.askTimeout")
+      .categories("client")
+      .version("0.3.0")
+      .doc("Timeout for LifecycleManager request reserve slots.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("30s")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2510]: 30s seems too shot, perhaps 60s?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:661]: DOne

**Pull request merged**